src/include/sof/audio/component.h

```diff
@@ -417,6 +421,20 @@ static inline int comp_params(struct comp_dev *dev,
 	return 0;
 }
 
+/**
+ * Fetch hardware stream parameters - only mandatory for DAI components.
+ * @param dev Component device.
+ * @param params hw parameters
+ * @return 0 if succeeded, error code otherwise.
+ */
+static inline int comp_dai_get_hw_params(struct comp_dev *dev,
+					 struct sof_ipc_stream_params *params)
+{
+	if (dev->drv->ops.dai_get_hw_params)
```

Comment:
Can we do a simple check here ```if (comp != DAI) return err```

Reply:
I referred to this in comment: https://github.com/thesofproject/sof/pull/2228#discussion_r367839760